manifests/init.pp

```diff
@@ -58,7 +58,7 @@
   Optional[Boolean] $unix_password_sync,
   Optional[String] $socket_options,
   Optional[String] $syslog,
-  Optional[Boolean] $ntlm_auth,
+  Variant[Enum['ntlmv1-permitted', 'ntlmv2-only', 'mschapv2-and-ntlmv2-only', 'disabled'], Boolean] $ntlm_auth,
```

Comment:
I think this could be Optional[Variant[Enum['ntlmv1-permitted', 'ntlmv2-only', 'mschapv2-and-ntlmv2-only', 'disabled'], Boolean]]?